verifier/src/evidence.rs

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! The full set of evidence needed for attesting a quote
+
+use crate::{Accessor, Advisories, Error, Result, TcbInfo, TcbInfoRaw};
+use der::DecodePem;
+use mc_sgx_core_types::{
+    Attributes, ConfigId, ConfigSvn, CpuSvn, ExtendedProductId, FamilyId, IsvProductId, IsvSvn,
+    MiscellaneousSelect, MrEnclave, MrSigner, ReportData,
+};
+use mc_sgx_dcap_types::{CertificationData, Quote3, TcbInfo as QuoteTcbInfo};
+use x509_cert::Certificate;
+
+/// The full set of evidence needed for verifying a quote
+///
+/// A wrapping container for a `Quote3` and a `TcbInfoRaw`. This can be used
+/// with the majority of the `Verifier` implementations from this crate.
+/// This allows one to compose one verifier and use an [`Evidence`] instance in
+/// the `verify()` method.
+///
+/// Importantly this will derive the [`Advisories`] related to the provided
+/// `quote` and `tcb_info_raw`, so that one can verify the allowed advisories.
+#[derive(Debug)]
+pub struct Evidence<'a, Q> {
```

Comment:
Up for suggestions on this name, it feels a bit too generic, but it may also be my Cisms creeping in and wanting to namespace the object name.